### ISSUE TITLE
fail2ban on ubuntu 18.04 does not seen to be working

### ISSUE THREAD

**Issue by knnleow** [fail2ban/fail2ban#2185]
### Environment:
                  Fail2Ban v0.11.0.dev2 running on ubuntu 18.04 latest patch 
                  Linux inwin01 4.15.0-24-generic #26-Ubuntu SMP

                  using the latest Fail2Ban from github bcos default OS Fail2Ban package also not working 

                  all the recommanded patched from 18.04

                  default setup not working.
                  enable sshd in jail.local
                  try default setting first, also change the jail.local entry --> "chain = INPUT"

### The issue:
              - install and configure jail.local
                # vi /etc/fail2ban/jail.local
                          : 
                       #chain = <known/chain>
                       chain = INPUT
                          : 
                       [sshd]
                       enabled = true
              - restart service and no error 
              - tail auth.log and fail2ban.log below
                       :  
  ==> auth.log <==
```
Jul 17 23:01:04 inwin01 systemd-logind[1095]: Removed session 12.
Jul 17 23:01:04 inwin01 sudo: pam_unix(sudo:session): session closed for user root
```
==> fail2ban.log <==
```
2018-07-17 23:01:04,923 fail2ban.server         [1208]: INFO    --------------------------------------------------
2018-07-17 23:01:04,923 fail2ban.server         [1208]: INFO    Starting Fail2ban v0.11.0.dev2
2018-07-17 23:01:04,923 fail2ban.observer       [1208]: INFO    Observer start...
2018-07-17 23:01:04,928 fail2ban.database       [1208]: INFO    Connected to fail2ban persistent database '/var/lib/fail2ban/fail2ban.sqlite3'
2018-07-17 23:01:04,928 fail2ban.jail           [1208]: INFO    Creating new jail 'sshd'
2018-07-17 23:01:04,934 fail2ban.jail           [1208]: INFO    Jail 'sshd' uses pyinotify {}
2018-07-17 23:01:04,937 fail2ban.jail           [1208]: INFO    Initiated 'pyinotify' backend
2018-07-17 23:01:04,937 fail2ban.filter         [1208]: INFO      maxLines: 1
2018-07-17 23:01:04,954 fail2ban.server         [1208]: INFO    Jail sshd is not a JournalFilter instance
2018-07-17 23:01:04,955 fail2ban.filter         [1208]: INFO    Added logfile: '/var/log/auth.log' (pos = 232099, hash = 55413f2ed7023394e25d5f2d446d8d40)
2018-07-17 23:01:04,956 fail2ban.filter         [1208]: INFO      maxRetry: 5
2018-07-17 23:01:04,957 fail2ban.filter         [1208]: INFO      encoding: UTF-8
2018-07-17 23:01:04,957 fail2ban.filter         [1208]: INFO      findtime: 604800
2018-07-17 23:01:04,958 fail2ban.actions        [1208]: INFO      banTime: 604800
2018-07-17 23:01:04,959 fail2ban.jail           [1208]: INFO    Jail 'sshd' started
```                 
              - simulate a ssh hack login waited for more than 20 tries, still SSH rules does not kick in
```
2018-07-17 23:01:04,956 fail2ban.filter         [1208]: INFO      maxRetry: 5
2018-07-17 23:01:04,957 fail2ban.filter         [1208]: INFO      encoding: UTF-8
2018-07-17 23:01:04,957 fail2ban.filter         [1208]: INFO      findtime: 604800
2018-07-17 23:01:04,958 fail2ban.actions        [1208]: INFO      banTime: 604800
2018-07-17 23:01:04,959 fail2ban.jail           [1208]: INFO    Jail 'sshd' started
```
==> auth.log <==
```
Jul 17 23:03:05 inwin01 sshd[1296]: Connection closed by authenticating user root 10.0.0.72 port 46038 [preauth]
Jul 17 23:03:08 inwin01 sshd[1298]: Connection closed by authenticating user root 10.0.0.72 port 46040 [preauth]
Jul 17 23:03:12 inwin01 sshd[1300]: Connection closed by authenticating user root 10.0.0.72 port 46042 [preauth]
Jul 17 23:03:15 inwin01 sshd[1302]: Connection closed by authenticating user root 10.0.0.72 port 46044 [preauth]
Jul 17 23:03:18 inwin01 sshd[1304]: Connection closed by authenticating user root 10.0.0.72 port 46048 [preauth]
Jul 17 23:03:21 inwin01 sshd[1306]: Connection closed by authenticating user root 10.0.0.72 port 46050 [preauth]
Jul 17 23:03:24 inwin01 sshd[1308]: Connection closed by authenticating user root 10.0.0.72 port 46052 [preauth]
Jul 17 23:03:27 inwin01 sshd[1310]: Connection closed by authenticating user root 10.0.0.72 port 46054 [preauth]
Jul 17 23:03:30 inwin01 sshd[1312]: Connection closed by authenticating user root 10.0.0.72 port 46058 [preauth]
Jul 17 23:03:33 inwin01 sshd[1314]: Connection closed by authenticating user root 10.0.0.72 port 46062 [preauth]
```
             - check iptables -L -v and no Fail2ban-rules added to INPUT chain
```
iptables -L -v |more 
Chain INPUT (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
45774 5378K ufw-before-logging-input  all  --  any    any     anywhere             anywhere            
45774 5378K ufw-before-input  all  --  any    any     anywhere             anywhere            
  259 33527 ufw-after-input  all  --  any    any     anywhere             anywhere            
    7   420 ufw-after-logging-input  all  --  any    any     anywhere             anywhere            
    7   420 ufw-reject-input  all  --  any    any     anywhere             anywhere            
    7   420 ufw-track-input  all  --  any    any     anywhere             anywhere            

Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DOCKER-USER  all  --  any    any     anywhere             anywhere            
    0     0 DOCKER-ISOLATION  all  --  any    any     anywhere             anywhere            
    0     0 ACCEPT     all  --  any    docker0  anywhere             anywhere             ctstate RELATED,ESTABLISHED
    0     0 DOCKER     all  --  any    docker0  anywhere             anywhere            
    0     0 ACCEPT     all  --  docker0 !docker0  anywhere             anywhere            
    0     0 ACCEPT     all  --  docker0 docker0  anywhere             anywhere            
    0     0 ufw-before-logging-forward  all  --  any    any     anywhere             anywhere            
    0     0 ufw-before-forward  all  --  any    any     anywhere             anywhere            
    0     0 ufw-after-forward  all  --  any    any     anywhere             anywhere            
    0     0 ufw-after-logging-forward  all  --  any    any     anywhere             anywhere            
    0     0 ufw-reject-forward  all  --  any    any     anywhere             anywhere            
    0     0 ufw-track-forward  all  --  any    any     anywhere             anywhere            

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
              : 
```
Note; there in UFW firewall running. this is the same as my working configuration running ubuntu 16.04 and  Fail2Ban  version v0.9.3 out of the box from ubuntu.

**Comment by sebres** [comment]
**TLDR** Could you please explain with few words what **exactly** the issue is?
(The issue template is pretty clear, but each tries to use own).

**Comment by sebres** [comment]
> Connection closed by authenticating user root 

It's a secondary message (that indeed will be not captured as failure by fail2ban). Similar messages will be used to find the IP address, if primary message with same session-id (qualified as failure) doesn't contain IP-address.
So it looks like:
```
Nov 23 21:50:19 sshd[8148]: Disconnecting: Too many authentication failures for root [preauth]
Nov 23 21:50:37 sshd[8148]: Connection closed by 192.0.2.1 [preauth]
```
So it looks like the sshd-logging on your system is not fulfilled - possibly some messages are filtered or going into different log-file (see distribution rules of rsyslog) or even into systemd-journal (so you should change the backend of this jails to systemd)

All the messages that will be currently captured by fail2ban could be found in
https://github.com/fail2ban/fail2ban/blob/0.11/fail2ban/tests/files/logs/sshd
Note only lines having `"match": true` in the failJSON-line above.

Do you really not see any log-line in some other logs resp. systemd-journal?

BTW. we've indeed some additional modes (`aggressive` or `ddos`) where this message should be captured as failure, but it does not because of the part `user root` in-between.
I'll fix this for the next release.
